File: src/Java/BehavioralPatterns/Mediator/CommunicationHub/src/Participants/AccelerometerSensor.java
package Participants;

import java.util.Random;

public class AccelerometerSensor extends Sensor {
    public AccelerometerSensor() {
        super("acceleration");
    }

    @Override
    public void receive(Participant sender, Object args) {
        if (args.equals("measure")) {
            setLastValue(new Random().nextDouble());
            getMediator().notify(this, "measure");
            getMediator().notify(this, getLastValue());
        }
    }

    @Override
    public void valueChanged(Object value) {
        setLastValue(value);
        getMediator().notify(this, "measure");
        getMediator().notify(this, getLastValue());
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/CommunicationHub/participants/accelerometerSensor.js

const Sensor = require("./sensor");

class AccelerometerSensor extends Sensor {
  constructor() {
    super("acceleration");
  }

  receive(sender, args) {
    if (args === "measure") {
      this.lastValue = Math.random();
      this.mediator.notify(this, "measure");
      this.mediator.notify(this, this.lastValue);
    }
  }

  valueChanged(value) {
    this.lastValue = value;
    this.mediator.notify(this, "measure");
    this.mediator.notify(this, this.lastValue);
  }
}

module.exports = AccelerometerSensor;
